src/lib/rng/jitter_rng/jitter_rng.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+* CPU Jitter Random Number Generator
+* (C) 2024 Planck Security S.A.
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/jitter_rng.h>
+
+#include <jitterentropy.h>
+
+namespace Botan {
+
+namespace {
+
+struct Jitter_RNG_Internal {
+      Jitter_RNG_Internal();
+      ~Jitter_RNG_Internal();
+      void collect_into_buffer(std::span<uint8_t> buf);
+
+   private:
+      rand_data* m_rand_data;
+};
+
+Jitter_RNG_Internal::Jitter_RNG_Internal() {
+   static int result = jent_entropy_init();
+
+   if(result != 0) {
+      // no further details documented regarding the return value
+      throw Internal_Error("JitterRNG: Can not be used");
+   }
+
+   constexpr unsigned int oversampling_rate = 0;  // use default oversampling
+   constexpr unsigned int flags = 0;
+
+   m_rand_data = jent_entropy_collector_alloc(oversampling_rate, flags);
+   if(!m_rand_data) {
+      throw Internal_Error("JitterRNG: Jitter entropy collector could not be allocated");
+   }
```

Comment:
```suggestion
   BOTAN_ASSERT_NONNULL(m_rand_data);
```

... at the expense of the more detailed error message, of course. But the assertion will state the code location in the error message which is just as helpful in debugging. Just a suggestion, please feel free to leave as is.